backend/core/src/main/java/org/sonarsource/sonarlint/core/fs/ClientFileSystemService.java

```diff
@@ -140,6 +144,7 @@ public void didUpdateFileSystem(DidUpdateFileSystemParams params) {
       if (clientFile != null) {
         filesByConfigScopeIdCache.get(clientFile.getConfigScopeId()).remove(uri);
         removed.add(clientFile);
+        readThroughFileCache.remove(Path.of(uri), clientFile.getCharset());
```

Comment:
A URI doesn't necessarily represent a local path, so this conversion might throw. Usually we try to reference files via their URI in the file system service, couldn't you do the same here?